src/core/lib/security/authorization/rbac_policy.cc

```diff
@@ -29,22 +29,46 @@ namespace grpc_core {
 // Rbac
 //
 
-Rbac::Rbac(Rbac::Action action, std::map<std::string, Policy> policies)
-    : action(action), policies(std::move(policies)) {}
+Rbac::Rbac(Rbac::Action action, std::map<std::string, Policy> policies,
+           absl::string_view name)
+    : action(action), policies(std::move(policies)), name(name) {}
 
 Rbac::Rbac(Rbac&& other) noexcept
-    : action(other.action), policies(std::move(other.policies)) {}
+    : action(other.action),
+      audit_condition(other.audit_condition),
+      policies(std::move(other.policies)),
+      logger_configs(std::move(other.logger_configs)),
+      name(std::move(other.name)) {}
 
 Rbac& Rbac::operator=(Rbac&& other) noexcept {
   action = other.action;
+  audit_condition = other.audit_condition;
   policies = std::move(other.policies);
+  logger_configs = std::move(other.logger_configs);
+  name = std::move(other.name);
   return *this;
 }
 
 std::string Rbac::ToString() const {
   std::vector<std::string> contents;
+  std::string condition_str;
+  switch (audit_condition) {
+    case Rbac::AuditCondition::kNone:
+      condition_str = "None";
+      break;
+    case AuditCondition::kOnDeny:
+      condition_str = "OnDeny";
+      break;
+    case AuditCondition::kOnAllow:
+      condition_str = "OnAllow";
+      break;
+    case AuditCondition::kOnDenyAndAllow:
+      condition_str = "OnDenyAndAllow";
+      break;
+  }
   contents.push_back(absl::StrFormat(
-      "Rbac action=%s{", action == Rbac::Action::kAllow ? "Allow" : "Deny"));
+      "Rbac name=%s action=%s audit_condition=%s{", name,
```

Comment:
Would it make sense to change the order of `name` and `action` here to mimic the previous structure?

Reply:
I feel name should always be the first thing if available. And the resulting string is gonna be different from before anyway, i.e. we are are falling back to existing behavior if name is absent or so.